### ISSUE TITLE
CONTRIBUTING.md: add details of how to handle Git workflow

### ISSUE THREAD

**Issue by rjhintz** [open-guides/og-aws#313]
CONTRIBUTING.md should probably include details of proper Git workflow to minimize extraneous commits and facilitate merges.

I'd propose Chef's [Community Contributions](https://docs.chef.io/community_contributions.html#community-contributions) as a good example to adapt.

**Comment by lynnlangit** [comment]
I agree, but think we should write our own simplified version of this

**Comment by rjhintz** [comment]
Added some proposed language as a [Quip doc](https://og-aws.quip.com/Bv2mAVanO75j)
